login/service_blackbox_test.go

```diff
@@ -0,0 +1,198 @@
+package login_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app"
+	. "github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+)
+
+var db *gorm.DB
+var loginService Service
+
+func TestMain(m *testing.M) {
+	if _, c := os.LookupEnv(resource.Database); c == false {
+		fmt.Printf(resource.StSkipReasonNotSet+"\n", resource.Database)
+		return
+	}
+
+	dbhost := os.Getenv("ALMIGHTY_DB_HOST")
+	if "" == dbhost {
+		panic("The environment variable ALMIGHTY_DB_HOST is not specified or empty.")
+	}
+	var err error
+	db, err = gorm.Open("postgres", fmt.Sprintf("host=%s database=postgres user=postgres password=mysecretpassword sslmode=disable", dbhost))
+	if err != nil {
+		panic("failed to connect database: " + err.Error())
+	}
+	defer db.Close()
+
+	oauth := &oauth2.Config{
+		ClientID:     "875da0d2113ba0a6951d",
+		ClientSecret: "2fe6736e90a9283036a37059d75ac0c82f4f5288",
+		Scopes:       []string{"user:email"},
+		Endpoint:     github.Endpoint,
+	}
+
+	publicKey, err := token.ParsePublicKey(token.RSAPublicKey)
+	if err != nil {
+		panic(err)
+	}
+
+	privateKey, err := token.ParsePrivateKey(token.RSAPrivateKey)
+	if err != nil {
+		panic(err)
+	}
+
+	tokenManager := token.NewManager(publicKey, privateKey)
+	userRepository := account.NewUserRepository(db)
+	identityRepository := account.NewIdentityRepository(db)
+	loginService = NewGitHubOAuth(oauth, identityRepository, userRepository, tokenManager)
+
+	os.Exit(m.Run())
+}
+
+func TestGithubOAuthAuthorizationRedirect(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	rw := httptest.NewRecorder()
+	u := &url.URL{
+		Path: fmt.Sprintf("/api/login/authorize"),
+	}
+	req, err := http.NewRequest("GET", u.String(), nil)
+	req.Header.Add("referer", "localhost")
+	if err != nil {
+		panic("invalid test " + err.Error()) // bug
+	}
+	prms := url.Values{}
+	ctx := context.Background()
+	goaCtx := goa.NewContext(goa.WithAction(ctx, "LoginTest"), rw, req, prms)
+	authorizeCtx, err := app.NewAuthorizeLoginContext(goaCtx, goa.New("LoginService"))
+	if err != nil {
+		panic("invalid test data " + err.Error()) // bug
+	}
+
+	err = loginService.Perform(authorizeCtx)
+
+	assert.Equal(t, 307, rw.Code)
+	assert.Contains(t, rw.Header().Get("Location"), "https://github.com/login/oauth/authorize")
+}
+
+func TestValidOAuthAuthorizationCode(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	// Current the OAuth code is generated as part of a UI workflow.
```

Comment:
Excellent that you create 'empty' tests to describe what should be tested, but please add a t.Fail("Not Implemented") or something so we don't forget to implement them :)